scripts/profiler/rtt_nordic_profiler_host.py

```diff
@@ -100,26 +97,28 @@ def shutdown(self):
     def disconnect(self):
         self.stop_logging_events()
         # read remaining data to buffer
-        try:
-            buf = self.jlink.rtt_read(self.config['rtt_data_channel'],
-                                      self.config['rtt_read_chunk_size'],
-                                      encoding=None)
+        while True:
+            try:
+                buf = self.jlink.rtt_read(self.config['rtt_data_channel'],
+                                          self.config['rtt_read_chunk_size'],
+                                          encoding=None)
+
+            except Exception:
```

Comment:
It's good to convert this so it doesn't catch system-exiting exceptions, but is there a more specific exception type that can be used here (like OSError or RuntimeError, or both) and elsewhere in this file?

Reply:
According to Python documentation: all built-in, non-system-exiting exceptions are derived from Exception class. So we will not catch the system-exiting exceptions here.

Also I assumed that we want to act like that if anything goes wrong when calling the function in the try block.

Reply:
Right, I am saying that changing to Exception is good exactly *because* it catches non-system-exiting exceptions. So this is an improvement.

My question is if we can narrow the exception types caught even more, so that we only catch the exceptions we expect this call to generate, allowing other types of exceptions to bubble up. Getting stack traces for unexpected exceptions is often useful for debugging, of course.

If you really want to catch everything, OK.

Reply:
I may switch to `APIErrror`, since this is exception raised by pynrfjprog. The reading function should not raise any other exception types I think. Thank you for the feedback.